packages/edit-site/src/components/block-preview/index.js

```diff
@@ -0,0 +1,167 @@
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import apiFetch from '@wordpress/api-fetch';
+import { serialize } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import EditorStyles from './editor-styles';
+import Iframe from './iframe';
+import { store as editSiteStore } from '../../store';
+
+const MAX_HEIGHT = 2000;
+
+function ScaledBlockPreview( {
+	viewportWidth,
+	containerWidth,
+	minHeight,
+	html,
+	additionalStyles = [],
+} ) {
+	if ( ! viewportWidth ) {
+		viewportWidth = containerWidth;
+	}
+
+	const [ contentResizeListener, { height: contentHeight } ] =
+		useResizeObserver();
+	const { styles } = useSelect( ( select ) => {
+		const settings = select( blockEditorStore ).getSettings();
+		return {
+			styles: settings.styles,
+		};
+	}, [] );
+
+	// Avoid scrollbars for pattern previews.
+	const editorStyles = useMemo( () => {
+		if ( styles ) {
+			return [
+				...styles,
+				{
+					css: 'body{height:auto;overflow:hidden;border:none;padding:0;}',
+					__unstableType: 'presets',
+				},
+				...additionalStyles,
+			];
+		}
+
+		return styles;
+	}, [ styles, additionalStyles ] );
+
+	const scale = containerWidth / viewportWidth;
+	const aspectRatio = contentHeight
+		? containerWidth / ( contentHeight * scale )
+		: 0;
+	return (
+		<Disabled
+			className="block-editor-block-preview__content"
+			style={ {
+				transform: `scale(${ scale })`,
+				// Using width + aspect-ratio instead of height here triggers browsers' native
+				// handling of scrollbar's visibility. It prevents the flickering issue seen
+				// in https://github.com/WordPress/gutenberg/issues/52027.
+				// See https://github.com/WordPress/gutenberg/pull/52921 for more info.
+				aspectRatio,
+				maxHeight:
+					contentHeight > MAX_HEIGHT ? MAX_HEIGHT * scale : undefined,
+				minHeight,
+			} }
+		>
+			<Iframe
+				contentRef={ useRefEffect( ( bodyElement ) => {
+					const {
+						ownerDocument: { documentElement },
+					} = bodyElement;
+					documentElement.classList.add(
+						'block-editor-block-preview__content-iframe'
+					);
+					documentElement.style.position = 'absolute';
+					documentElement.style.width = '100%';
+
+					// Necessary for contentResizeListener to work.
+					bodyElement.style.boxSizing = 'border-box';
+					bodyElement.style.position = 'absolute';
+					bodyElement.style.width = '100%';
+				}, [] ) }
+				aria-hidden
+				tabIndex={ -1 }
+				style={ {
+					position: 'absolute',
+					width: viewportWidth,
+					height: contentHeight,
+					pointerEvents: 'none',
+					// This is a catch-all max-height for patterns.
+					// See: https://github.com/WordPress/gutenberg/pull/38175.
+					maxHeight: MAX_HEIGHT,
+					minHeight:
+						scale !== 0 && scale < 1 && minHeight
+							? minHeight / scale
+							: minHeight,
+				} }
+			>
+				<EditorStyles styles={ editorStyles } />
+				{ contentResizeListener }
+				<div
```

Comment:
Is this div the only thing that is different compared to the default block preview? If that's the case, why not just make the `blockPreview` setting return just this part and not the whole thing and just reuse the `BlockPreview` except this part.

Reply:
I'll review this - I think initially I thought there would be more simplification. I'll see if there is more if not I'll go with your observations. 🙏